### PR TITLE
Add pp that computes the total volume of the domain

### PR DESCRIPTION
**Pull request by anne-glerum**
This PR adds a statistics postprocessor that computes the volume of the computational domain. This can be useful when you apply certain in- and outflow boundary conditions in combination with a free surface or other type of mesh deformation and you want to check that volume is conserved.

@Djneu I noticed that you wrote a plugin that also outputs the domain volume, but as this a one-liner, I thought this implementation would be the preferred one.

### For all pull requests:

* [x] I have followed the [instructions for indenting my code](../blob/main/CONTRIBUTING.md#making-aspect-better).

### For new features/models or changes of existing features:

* [x] I have tested my new feature locally to ensure it is correct.
* [x] I have [created a testcase](http://www.math.clemson.edu/~heister/manual.pdf#sec%3Awriting_tests) for the new feature/benchmark in the [tests/](../blob/main/tests/) directory.
* [x] I have added a changelog entry in the [doc/modules/changes](../blob/main/doc/modules/changes) directory that will inform other users of my change.

### CODE REVIEW

File: include/aspect/postprocess/domain_volume_statistics.h [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+
+    /**
+     * A postprocessor that computes some statistics about the material_flux
+     * in and out of boundaries.
```
Comment (gassmoeller) [inline include/aspect/postprocess/domain_volume_statistics.h:36]: This needs an update

File: source/postprocess/domain_volume_statistics.cc [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/postprocess/domain_volume_statistics.h>
+#include <aspect/utilities.h>
+#include <aspect/geometry_model/interface.h>
```
Comment (gassmoeller) [inline source/postprocess/domain_volume_statistics.cc:24]: These are not needed I think.

File: source/postprocess/domain_volume_statistics.cc [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/fe/fe_values.h>
```
Comment (gassmoeller) [inline source/postprocess/domain_volume_statistics.cc:27]: These can also be removed.

File: source/postprocess/domain_volume_statistics.cc [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    template <int dim>
+    std::pair<std::string,std::string>
+    DomainVolume<dim>::execute (TableHandler &statistics)
+    {
+      const double global_volume = GridTools::volume (this->get_triangulation(), this->get_mapping());
+
+      // add the volume to the statistics object
+      // and create a single string that can be output to the screen
+      std::ostringstream screen_text;
+      const std::string name = "Model domain volume (m3) ";
```
Comment (gassmoeller) [inline source/postprocess/domain_volume_statistics.cc:43]: Can you make the unit dependent on dimension? Here and in line 53.

**Review by gassmoeller — APPROVED** [review]
Very useful! Just a few comments, and please add a changelog entry.

**Comment by gassmoeller** [comment]
After checking: Actually, the Simulator class already has a variable `global_volume`, which is updated after ever mesh deformation. Can you instead of computing it again use the SimulatorAccess function `get_volume` to access the existing variable? Would save a bit of time.

File: source/postprocess/domain_volume_statistics.cc [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    template <int dim>
+    std::pair<std::string,std::string>
+    DomainVolume<dim>::execute (TableHandler &statistics)
+    {
+      // const double global_volume = this->get_volume();
+      const double global_volume = GridTools::volume(this->get_triangulation(), this->get_mapping());
+
+      // add the volume to the statistics object
+      // and create a single string that can be output to the screen
+      std::ostringstream screen_text;
```
Comment (tjhei) [inline source/postprocess/domain_volume_statistics.cc:39]: Move the declaration down to where it is used first.

**Comment by tjhei** [comment]
> Hence the global volume in the simulator is not correct.

Well, I think we should look at that as well and fix it.

**Comment by bangerth** [comment]
@anne-glerum Can you rebase this patch and try again? With @gassmoeller 's patch in, this might now actually work!

**Comment by anne-glerum** [comment]
With #4881 the `simulator_access` function `get_volume` now returns the correct value of the domain volume over time. I'll update the test results when the tester is done.

File: source/postprocess/domain_volume_statistics.cc [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  {
+    template <int dim>
+    std::pair<std::string,std::string>
+    DomainVolume<dim>::execute (TableHandler &statistics)
+    {
+      // Retrieve the current domain volume
+      const double global_volume = this->get_volume();
+
+      // add the volume to the statistics object
+      const std::string unit = (dim == 2) ? "m2" : "m3";
```
Comment (bangerth) [inline source/postprocess/domain_volume_statistics.cc:38]:
```suggestion
      const std::string unit = (dim == 2) ? "m^2" : "m^3";
```

**Review by bangerth — COMMENTED** [review]
OK if you fix the one string!

**Comment by tjhei** [comment]
You need to update the output of ``free_surface_tangential_mesh_velocity``, @anne-glerum

**Comment by anne-glerum** [comment]
Done!

**Review by gassmoeller — APPROVED** [review]
Thanks!

**Pull request merged**